### ISSUE TITLE
reordering while preparing for output

### ISSUE THREAD

**Issue by avocadit** [charnley/rmsd#93]
Hi,
using the module I got geometries which, in my humble opinion, are too far from each other for a very small RMSD calculated. These molecules look similar if I take original geometries.
When the --reorder option is turned on, the q_coord list is reordered (lines 2021-2022). Aren't the lines (2046-2047)
```
        if q_review is not None:
            q_coord = q_coord[q_review]
            q_all_atoms = q_all_atoms[q_review]
```
changing their order again, when output is requested? When I commented out the mentioned part, the new geometries look more aligned with each other and my expectations.

**Comment by nbehrnd** [comment]
> Hi,
> using the module I got geometries which, in my humble opinion, are too far from each other for a very small RMSD calculated.

Do you have representative input/output data to substantiate/illustrate your
findings with `calculate_rmsd`?  Do your findings relate to the most recent
commit of `calculate_rmsd` (if not which one)?  If the alignment and
corresponding RMSD seem to be off, what is the complete instruction delivered
to `calculate_rmsd`? Did you run a comparison with other implementations of the
Kabsch test (e.g., pymol, Jmol) and can you provide their output geometry and
RMSD as a benchmark?

**Comment by avocadit** [comment]
> Hi, using the module I got geometries which, in my humble opinion, are too far from each other for a very small RMSD calculated.
> Do you have representative input/output data to substantiate/illustrate your findings with `calculate_rmsd`? Do your findings relate to the most recent commit of `calculate_rmsd` (if not which one)? If the alignment and corresponding RMSD seem to be off, what is the complete instruction delivered to `calculate_rmsd`? Did you run a comparison with other implementations of the Kabsch test (e.g., pymol, Jmol) and can you provide their output geometry and RMSD as a benchmark?

I have two conformers (a.xyz, b.xyz) of a small molecule which have very similar total energies, so I started to believe they may be practically the same. Eyeballing them in Avogadro did not undermine this conclusion.
`calculate_rmsd --reorder a.xyz b.xyz` gave me RMSD = 0.00059 Å for this pair of geometries, while for some other pairs of conformers it was around 1 or 2 Å. So I thought the case was closed, but using `calculate_rmsd --reorder --print a.xyz b.xyz` to get c.xyz and plotting a.xyz with c.xyz in matplotlib (a_c.jpg) showed the distances, which cannot give such small RMSD.
I started checking whats happening with the positions in the calculate_rmsd, printing q_coord and I see that the order of atoms change back and forth in the lines:
```
    elif settings.reorder:
        assert reorder_method is not None, "Cannot reorder without selecting --reorder method"
        q_review = reorder_method(p_atoms, q_atoms, p_coord, q_coord)
        q_coord = q_coord[q_review]
        q_atoms = q_atoms[q_review]
        assert all(
            p_atoms == q_atoms
        ), "error: Structure not aligned. Please submit bug report at http://github.com/charnley/rmsd"
    # print result
    if settings.output:
        # ...
        if q_swap is not None:
            q_coord = q_coord[:, q_swap]
        if q_reflection is not None:
            q_coord = np.dot(q_coord, np.diag(q_reflection))
        q_coord -= centroid(q_coord)
        if q_review is not None:
            q_coord = q_coord[q_review]
            q_all_atoms = q_all_atoms[q_review]
```
In both reordering episodes the same mask q_review is used.
After commenting out the second reordering, `calculate_rmsd --reorder --print a.xyz b.xyz` produced d.xyz which has different positions (a_d.jpg) than c.xyz, but much more consistent with the value of 0.00059 Å.

I didn't have time to make a comparison with other implementations. There is obviously a chance I didn't contemplated over the issue enough or I make some mistake -- sorry for taking up your time if that's the case.

Mentioned files: [rmsd_tst.tar.gz](https://github.com/charnley/rmsd/files/10244843/rmsd_tst.tar.gz)

**Comment by charnley** [comment]
Thanks @avocadit, looks like you are correct. I created a new branch/test to capture it and will fix it ASAP.

**Comment by charnley** [comment]
@avocadit , can confirm there was an inconsistency in the reorder and print setup. Applying the reorder was inconsistent when using reflections and not, so it was uncaught in the tests. A test has been added to capture your case, and the bug has been fixed (#94)
